src/QuadForm/Lattices.jl

```diff
@@ -966,6 +966,45 @@ function restrict_scalars(L::AbsLat)
   return ZLat(Vabs, Mabs)
 end
 
+#=
+    _restrict_scalars_with_map(L) -> 
+      Tuple{ZLat, Hecke.VecSpaceRes{Hecke.NfRel{nf_elem}, Hecke.NfRelElem{nf_elem}}}
+
+Takes a lattice $L$ and returns a $\mathbb{Z}$-Lattice and the restriction map $f$.
+=#
+function _restrict_scalars_with_map(L)
+  V = ambient_space(L)
+  Vabs, f = restrict_scalars(V, FlintQQ)
+  Babs = absolute_basis(L)
+  Mabs = zero_matrix(FlintQQ, length(Babs), rank(Vabs))
+  for i in 1:length(Babs)
+    v = f\(Babs[i])
+    for j in 1:length(v)
+      Mabs[i, j] = v[j]
+    end
+  end
+  return ZLat(Vabs, Mabs), f
+end
+
+#=
+    _restrict_scalars_with_respect_to_map(L, f, Vabs)(L) -> 
+      Tuple{ZLat, Hecke.VecSpaceRes{Hecke.NfRel{nf_elem}, Hecke.NfRelElem{nf_elem}}}
+
+Takes a lattice $L$, the restriction map $f$ and returns the $\mathbb{Z}$-lattice 
+obtained by restricting scalars.
+=#
+function _restrict_scalars_with_respect_to_map(L, f, Vabs)
+  Babs = absolute_basis(L)
+  Mabs = zero_matrix(FlintQQ, length(Babs), rank(Vabs))
+  for i in 1:length(Babs)
+    v = f\(Babs[i])
+    for j in 1:length(v)
+      Mabs[i, j] = v[j]
+    end
+  end
+  return ZLat(Vabs, Mabs)
+end
+
```

Comment:
I guess those two functions can be made open to the large public, not only internal, with their own documentation and tests, and split in another PR too. Maybe the second one can drop the `map` at the end of the name and keep it `restriction_of_scalars_with_respect_to`.